### PR TITLE
feat: add possibility to set rename as a function

### PR DESCRIPTION
**Pull request by Jisay**
## Description

Support rename argument as a function (or a string).

## Links

- #19

### CODE REVIEW

File: src/options.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export type Target = {
   /**
    * rename
    */
-  rename?: string
+  rename?: string | (() => string)
```
Comment (sapphi-red) [inline src/options.ts:37]: I don't think this type is correct.

File: src/utils.ts [hunk 2]
```diff
@@ -14,6 +14,14 @@ export type SimpleTarget = {
   preserveTimestamps: boolean
 }
 
+function renameTarget(target, rename, src) {
```
Comment (sapphi-red) [inline src/utils.ts:17]: Would you add a type to this function?

File: src/utils.ts [hunk 3]
```diff
@@ -14,6 +14,14 @@ export type SimpleTarget = {
...
+  const parsedPath = path.parse(target)
```
Comment (sapphi-red) [inline src/utils.ts:18]: Is this correct to pass `target`? I wonder if this should be `src`.

**Review by sapphi-red — CHANGES_REQUESTED** [review]
Thanks for the PR!
Would you add a test for this?

File: src/options.ts (hunk 1 above)
Comment (Jisay) [reply src/options.ts:37]: you're right. It needs some params in the function declaration.

File: src/utils.ts (hunk 3 above)
Comment (Jisay) [reply src/utils.ts:18]: I copy/paste it from `rollup-plugin-copy`  (https://github.com/vladshcherbin/rollup-plugin-copy/blob/507bf5e99aa2c6d0d858821e627cb7617a1d9a6d/src/index.js#L19-L25). I guess it's correct :-)

File: src/utils.ts (hunk 2 above)
Comment (Jisay) [reply src/utils.ts:17]: Sure

**Comment by Jisay** [comment]
I have added some test for rename option.
One as a simple string and one as a function.
Feel free to tell me if I added them at the right place :-)

**Review by sapphi-red — APPROVED** [review]

**Comment by sapphi-red** [comment]
@Jisay Thanks! I have released this as v0.10.0.

**Pull request merged**